requirements.txt

```diff
@@ -1,12 +0,0 @@
-# Core dependencies
-# `vtk` must be explicitly installed
-# It is critical that VTK is *not* listed here.
-#  This is to prevent installing `vtk` from PyPI
-#  when using a custom variant like `vtk_osmesa`
-#  which we do in some of our CI jobs.
```

Comment:
We can keep this file for this purpose if necessary. Though, at least for our CI jobs, in practice it seems that running 
```
pip install pyvista
pip uninstall vtk
pip install vtk-osmesa
```
is a more typical way to do this, and I think should be preferred over:
```
pip install --no-deps
pip install -r requirements.txt vtk-osmesa
```

The former is more explicit, though the latter is a bit wasteful with the unnecessary vtk install

If kept, it should probably be renamed (otherwise it may be ambigious if `pyproject.toml` or `requirements.txt` should be used)
Also, keeping this file would require bumping versions in two files instead of one...

Maybe `pip install pyvista-no-vtk` is best? https://github.com/pyvista/pyvista/pull/6361